airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/cdk/load/data/sql/SqlModels.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data.sql
+
+import java.sql.Types
+
+/**
+ * Representation of a colum in a SQL table.
+ *
+ * @param name The name of the column
+ * @param type The data type of the column (see [Types] for values).
+ * @param isPrimaryKey Whether the column represents a primary key.
+ * @param isNullable Whether the column's value supports null values.
+ */
+data class SqlColumn(
+    val name: String,
+    val type: Int,
+    val isPrimaryKey: Boolean = false,
+    val isNullable: Boolean = false
+)
+
+/**
+ * Representation of a SQL table.
+ *
+ * @param columns The list of columns in the table.
+ */
+data class SqlTable(val columns: List<SqlColumn>)
+
+/**
+ * Representation of a value in a SQL row/column cell.
+ *
+ * @param name The name of the column.
+ * @param value The value of the row/column cell.
+ * @param type The SQL type of the row/column cell (see [Types] for values).
+ */
+data class SqlTableRowValue(val name: String, val value: Any?, val type: Int)
```

Comment:
I don't know if we want to be beholden to java.sql.Type - e.g. will this be a problem for bigquery?

(one of the old CDK mistakes was plumbing JDBC stuff too deeply into things, so we had to duplicate a lot of functionality in bigquery, and to some extent snowflake+redshift)

totally fair if this is intended to be mssql-specific though

Reply:
@edgao Good call out.  My thought was that this could be SQL-specific, not necessarily MSSQL specific.  The MSSQL JDBC Driver does provide its version of `Types.java`, which has a few more types not part of standard SQL.  If it makes sense to make this MSSQL specific instead of generic SQL specific, I can rename this and leverage the types from the JDBC driver too.  That would also mean that these classes would live in the destination and would not need to move to a CDK toolkit, as I doubt that we would have use for MSSQL converters anywhere else.  Let me know what you think/I'm happy to make this MSSQL specific and refactor later if we find general use for it.

Reply:
mssql specific feels right to me. My suspicion is this ends up being parameterized on some Types enum (kind of related to the other thread below, about postgres jsonb instead of blob, etc)

Reply:
I will refactor this to make them MSSQL specific

Reply:
I've moved this out of the CDK toolkit package.  We can refactor/parameterize when we do the next one if there is a reason to share/reuse.